core/class/sequencing.class.php

```diff
@@ -251,20 +251,28 @@ public function evaluateEachConditions($_type){ // $_type ici peut etre trigger
       // les infos du JS peuvent etre : trigger, trigger_prog, trigger_timerange, trigger_cancel, trigger_cancel_prog et trigger_cancel_timerange.
 
       $results = array(); // va stocker le resultat de toutes les conditions (oui on va recalculer notre trigger éventuel aussi...;-( ))
-      foreach ($this->getConfiguration($_type) as $triggerOrCond) {
-        $results[$triggerOrCond['name']] = $this->checkTriggerValues($triggerOrCond, false, $_type); // false : c'est pas un trigger qui appele (default), on veut juste lire
+      $triggers = $this->getConfiguration($_type);
+      if (count($triggers)>0){
+        foreach ($triggers as $triggerOrCond) {
+            $results[$triggerOrCond['name']] = $this->checkTriggerValues($triggerOrCond, false, $_type); // false : c'est pas un trigger qui appelle (default), on veut juste lire
+        }
       }
 
-      foreach ($this->getConfiguration($_type.'_timerange') as $triggerOrCond) {
-        $results[$triggerOrCond['name']] = $this->checkCondTimeRange($triggerOrCond);
+      $triggers_timerange = $this->getConfiguration($_type.'_timerange');
+      if (count($triggers_timerange)>0){
+        foreach (triggers_timerange as $triggerOrCond) {
```

Comment:
il manque un $ dans le foreach, non ?

Reply:
Hi !
Bien vu ! Oui il manque un $... A force on se croise les yeux
Les tests sont en cours, d'où le PR pour avoir les infos dans un coin. Je te fais un retour dès que possible
@+

Reply:
ok, alors je laisse le PR ouvert pour l'instant si t'as d'autres corrections à apporter et j'attends ton GO pour merger. 

merci beaucoup en tout cas ! ;-)

Reply:
Il n'y a pas de quoi. Il est sympa ce petit plugin

Reply:
Hello, tout est ok chez toi, je merge et passe en stable ?

Reply:
Salut
En fait c'est pas la bonne solution : en l'état c'est pas directement "countable". Et c'est pas non plus une chaine vide... 
Pour l'instant je sèche et j'ai pas trouvé de test correct...
Il faut que je creuse encore un peu mais pour l'instant je suis dans la pose de placo !
++

Reply:
;-)
Ya pas d'urgence de toute façon !
Et tu as essayé avec un "isset($trigger)" ?

Ben tu me dis quand tu as une bonne solution qui te conviens alors ! 

Moi je peux accoucher à tout instant (mais ça peut encore prendre au moins 3 semaines encore aussi...), donc si je te répond pas, c'est que j'ai d'autres priorités qui sont arrivées ! ;-)

A+

Reply:
Hé mais c'est une chouette nouvelle ça ! Félicitations (et bon courage)

Coté code, le isset() n'aide pas non plus, on a bien un retour à tous les coups ... 

Sauf trompage, ta classe utilse la fonction getConfiguration hérité de eqLogic =>
	public function getConfiguration($_key = '', $_default = '') {
		return utils::getJsonAttr($this->configuration, $_key, $_default);
	}

Qui elle exploite ce bout de code (bien degeu)
 	public static function getJsonAttr(&$_attr, $_key = '', $_default = '') {
		if (is_array($_attr)) {
			if ($_key == '') {
				return $_attr;
			}
		} else {
			if ($_key == '') {
				return is_json($_attr, array());
			}
			if ($_attr === '') {
				if (is_array($_key)) {
					foreach ($_key as $key) {
						$return[$key] = $_default;
					}
					return $return;
				}
				return $_default;
			}
			$_attr = json_decode($_attr, true);
		}
		if (is_array($_key)) {
			$return = array();
			foreach ($_key as $key) {
				$return[$key] = (isset($_attr[$key]) && $_attr[$key] !== '') ? $_attr[$key] : $_default;
			}
			return $return;
		}
		return (isset($_attr[$_key]) && $_attr[$_key] !== '') ? $_attr[$_key] : $_default;
	}

Je pensais que dans l’hypothèse où le trigger n'existait pas en base, on retournait $_default qui vaut '' d'après la primitive...
Mais ça n'a pas l'air d'aller....
Je pense que je vais coder une copie de cette fonction avec des traces et, en décomposant avec un exemple ça ira mieux. 

Avec un peu de chance j'aurai une soluce avant les 3 ans de ton futur bout de chou ^^

Reply:
Oui c'est une chouette nouvelle, mais bien fatiguant les dernières semaines ! 

Et si au lieu d'un count() tu testais que c'est un array ? 

Si on reprend l'erreur d'origine : 
`PHP Warning:  Invalid argument supplied for foreach() in /var/www/html/plugins/sequencing/core/class/sequencing.class.php on line 258
`

Or l'argument attendu pour le foreach, c'est juste un array
Extrait de la doc php : 
```
foreach (array_expression as $value){
    //commandes
}
foreach (array_expression as $key => $value){
    //commandes
}
```

Donc s'assurer que la conf remonte un array et ne faire tourner le foreach que si oui, ça devrait résoudre le truc, non ?

++

Reply:
Bien vu ! ça à l'air de bien marcher avec isArray()
Je réactive des séquences bidons pour bien passer dans les 3 cas et je met à jours le PR.
Il va falloir que je révise un truc quand même un array[] c'est normalement OK avec count()....

Courage de ton coté !
++